tiledb/array.py

```diff
@@ -243,3 +251,761 @@ def index_domain_subarray(array, dom, idx: tuple):
+        if not ctx:
+            ctx = default_ctx()
+
+        if "preloaded_array" not in kwargs:
+            self.array = preload_array(uri, mode, key, timestamp, ctx)
+        else:
+            self.array = kwargs["preloaded_array"]
+
+        # view on a single attribute
+        schema = self.array._schema()
+        if attr and not any(
+            attr == schema._attr(i)._name for i in range(schema._nattr)
+        ):
+            self.array._close()
+            raise KeyError("No attribute matching '{}'".format(attr))
+        else:
+            self.view_attr = attr if attr is not None else None
+
+        self.ctx = ctx
+        self.key = key
+        self.uri = uri
+        self.domain_index = DomainIndexer(self)
+        self.pyquery = None
+        self.__buffers = None
+
+        self.last_fragment_info = dict()
+        self.meta = Metadata(self)
+
+    def __capsule__(self):
+        return self.array.__capsule__()
+
+    def __repr__(self):
+        if self.isopen:
+            return "Array(type={0}, uri={1!r}, mode={2}, ndim={3})".format(
+                "Sparse" if self.schema.sparse else "Dense",
+                self.uri,
+                self.mode,
+                self.schema.ndim,
+            )
+        else:
+            return "Array(uri={0!r}, mode=closed)"
+
+    @classmethod
+    def create(cls, uri, schema, key=None, overwrite=False, ctx=None):
+        """Creates a TileDB Array at the given URI
+
+        :param str uri: URI at which to create the new empty array.
+        :param ArraySchema schema: Schema for the array
+        :param str key: (default None) Encryption key to use for array
+        :param bool overwrite: (default False) Overwrite the array if it already exists
+        :param Ctx ctx: (default None) Optional TileDB Ctx used when creating the array,
+                        by default uses the ArraySchema's associated context
+                        (*not* necessarily ``tiledb.default_ctx``).
+
+        """
+        if ctx and not isinstance(ctx, Ctx):
+            raise TypeError(
+                "tiledb.Array.create() expected tiledb.Ctx " "object to argument ctx"
+            )
+        if not ctx:
+            ctx = schema.ctx
+
+        from .dense_array import DenseArrayImpl
+        from .sparse_array import SparseArrayImpl
+
+        if issubclass(cls, DenseArrayImpl) and schema.sparse:
+            raise ValueError(
+                "Array.create `schema` argument must be a dense schema for DenseArray and subclasses"
+            )
+        if issubclass(cls, SparseArrayImpl) and not schema.sparse:
+            raise ValueError(
+                "Array.create `schema` argument must be a sparse schema for SparseArray and subclasses"
+            )
+
+        if key:
+            config = tiledb.Config(ctx.config())
+            config["sm.encryption_type"] = "AES_256_GCM"
+            config["sm.encryption_key"] = key
+            ctx = tiledb.Ctx(config)
+
+        if overwrite:
```

Comment:
```suggestion
        if overwrite is not None:
```

Reply:
`overwrite` is a boolean. I suggest keeping this condition unchanged.